src/Arcus.EventGrid/Parsers/EventGridParser.cs

```diff
@@ -62,27 +70,71 @@ public static EventGridEventBatch<TEvent> Parse<TEvent>(string rawJsonBody)
         /// <param name="rawJsonBody">Raw JSON body</param>
         /// <param name="sessionId">Session id for event grid message</param>
         public static EventGridEventBatch<TEvent> Parse<TEvent>(string rawJsonBody, string sessionId)
-            where TEvent : IEvent
+            where TEvent : class, IEvent
         {
             Guard.NotNullOrWhitespace(rawJsonBody, nameof(rawJsonBody));
             Guard.NotNullOrWhitespace(sessionId, nameof(sessionId));
 
-             var array = JArray.Parse(rawJsonBody);
-             var settings = new JsonSerializerSettings
-             {
-                 ConstructorHandling = ConstructorHandling.AllowNonPublicDefaultConstructor,
-             };
-
+            var array = JArray.Parse(rawJsonBody);
+            
             var deserializedEvents = new List<TEvent>();
             foreach (var eventObject in array.Children<JObject>())
             {
                 var rawEvent = eventObject.ToString();
-                var gridEvent = JsonConvert.DeserializeObject<TEvent>(rawEvent, settings);
+
+                var gridEvent = JsonConvert.DeserializeObject<TEvent>(rawEvent, JsonSerializerSettings);
                 deserializedEvents.Add(gridEvent);
             }
 
             var result = new EventGridEventBatch<TEvent>(sessionId, deserializedEvents);
             return result;
         }
+
+        /// <summary>
+        ///     Parses a string to a <see cref="EventGridEventBatch{TEvent}"/> from either a <see cref="CloudEvent"/> or <see cref="EventGridEvent"/> implementation.
+        /// </summary>
+        /// <param name="rawJsonBody">Raw JSON body</param>
+        public static EventGridEventBatch<CloudOrEventGridEvent> Parse(string rawJsonBody)
+        {
+            string sessionId = Guid.NewGuid().ToString();
+
+            var eventGridEventBatch = Parse(rawJsonBody, sessionId);
+            return eventGridEventBatch;
+        }
+
+        /// <summary>
+        ///     Parses a string to a <see cref="EventGridEventBatch{TEvent}"/> from either a <see cref="CloudEvent"/> or <see cref="EventGridEvent"/> implementation.
+        /// </summary>
+        /// <param name="rawJsonBody">Raw JSON body</param>
+        /// <param name="sessionId">Session id for event grid message</param>
+        public static EventGridEventBatch<CloudOrEventGridEvent> Parse(string rawJsonBody, string sessionId)
```

Comment:
Please refresh my memory, why do we need multiple methods?

Reply:
1. For custom events (with and without session ID)
2. Shortcut for Azure Event Grid events by passing only the event data type
2. Abstracted parsing (both Cloud & Event Grid Events supported)

Reply:
Sounds good - Thanks for refreshing!